pkg/skaffold/build/cluster/kaniko_test.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/testutil"
+)
+
+func TestEnvInterpolation(t *testing.T) {
+	imageStr := "why.com/is/this/such/a/long/repo/name/testimage:testtag"
+	artifact := &latest.KanikoArtifact{
+		Env: []v1.EnvVar{{Name: "hui", Value: "buh"}},
+	}
+	generatedEnvs, err := generateEnvFromImage(imageStr)
+	if err != nil {
+		t.Fatalf("error processing generated env variables from image uri: %s", err)
+	}
+	env, err := evaluateEnv(artifact.Env, generatedEnvs...)
```

Comment:
The documentation in `v2beta3.json` states that the env vars are available for `buildArgs:` but it seems that is it only the case for `env:`. Should I report a bug?